frontend/src/features/mapSideBar/SidebarContents/subforms/AddParentParcelsForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { useLocation, Link } from 'react-router-dom';
+import { getIn, useFormikContext } from 'formik';
+import React, { useEffect } from 'react';
+import { Col, Form } from 'react-bootstrap';
+import { Label } from 'components/common/Label';
+import { Input, DisplayError } from 'components/common/form';
+import { pidFormatter } from 'features/properties/components/forms/subforms/PidPinForm';
+import SearchButton from 'components/common/form/SearchButton';
+import { toast } from 'react-toastify';
+import _ from 'lodash';
+import { IParcel } from 'actions/parcelsActions';
+import { withNameSpace } from 'utils/formUtils';
+import queryString from 'query-string';
+import { ISteppedFormValues } from 'components/common/form/StepForm';
+import { ISearchFields } from '../LandForm';
+import { FaRegTimesCircle } from 'react-icons/fa';
+import dequal from 'dequal';
+import styled from 'styled-components';
+
+interface IAddParentParcelsFormProps {
+  /** used for determining nameSpace of field */
+  nameSpace?: any;
+  /** Function that searches for a parcel matching a pid within the API */
+  findMatchingPid?: (pid: string, nameSpace?: string | undefined) => Promise<IParcel | undefined>;
+  /** whether or not the fields on this form can be interacted with */
+  disabled?: boolean;
+}
+
+const ErrorMessage = styled(DisplayError)`
+  display: block !important;
+`;
+
+/**
+ * Component that displays a search field that searches and adds valid pids, and allows the user to manage this list.
+ * @param {IAddParentParcelsFormProps} param0
+ */
+const AddParentParcelsForm = ({
+  nameSpace,
+  findMatchingPid,
+  disabled,
+}: IAddParentParcelsFormProps) => {
+  const { values, initialValues, touched, setFieldValue, setFieldTouched } = useFormikContext<
+    ISteppedFormValues<IParcel & ISearchFields>
+  >();
+  const location = useLocation();
+  const parcels = getIn(values, withNameSpace(nameSpace, 'parcels'));
+  const initialParcels = getIn(initialValues, withNameSpace(nameSpace, 'parcels'));
+  const touch = getIn(touched, withNameSpace(nameSpace, 'parcels'));
+
+  useEffect(() => {
+    if (!dequal(initialParcels, parcels)) {
+      setFieldTouched(withNameSpace(nameSpace, 'parcels') || touch);
+    }
+  }, [initialParcels, nameSpace, parcels, setFieldTouched, touch]);
+  return (
+    <>
+      <Col md={12}>
+        <h5>Parent Parcels</h5>
+      </Col>
+      <Col md={12}>
+        <Form.Row>
+          <Label>Pick Parent Parcel(s)</Label>
+          <Input
+            displayErrorTooltips
+            className="input-small"
+            disabled={disabled}
+            pattern={RegExp(/^[\d\- ]*$/)}
```

Comment:
oof I need to relearn my regular expressions apparently

Reply:
https://regexr.com/ helps